src/cider/nrepl/middleware/util/java.clj

```diff
@@ -73,9 +73,19 @@
        (str/replace "$" ".")
        (str ".html")))
   ([class member argtypes]
-   (str (javadoc-url class) "#" member
-        (when argtypes
-          (str "(" (str/join ",%20" argtypes) ")")))))
+   (let [java-version (Integer/parseInt util/java-api-version)
+         old-format   (< java-version 8)
+         open-paren   (if old-format "(" "-")
```

Comment:
Or one `if` branching out to two helper functions. :-) Anyways, we're on the same page here.